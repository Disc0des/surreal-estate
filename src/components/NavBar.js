import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import logo from "../estate-logo.png";
import "../styles/NavBar.css";

function NavBar() {
  const [userId, setUserId] = useState("");

  const onLogin = (response) => {
    setUserId(response.userID);
  };

  const onLogout = () => {
    window.FB.logout(setUserId(""));
  };

  return (
    <div className="navbar">
      <ul className="navbar-links">
        <li>
          <img src={logo} alt="logo" className="logo" />
        </li>
        <NavLink
          exact
          to="/"
          activeClassName="selected-link"
          className="navbar-links-item"
        >
          <li>View Properties</li>
        </NavLink>
        <NavLink
          to="/add-property"
          activeClassName="selected-link"
          className="navbar-links-item"
        >
          <li>Add Property</li>
        </NavLink>
      </ul>
      {(userId && (
        <button type="submit" className="sign-out" onClick={onLogout}>
          Sign Out
        </button>
      )) || (
        <FacebookLogin
          appId="279855101021290"
          callback={onLogin}
          icon="fa-facebook"
          textButton="   Login"
        />
      )}
    </div>
  );
}

export default NavBar;
