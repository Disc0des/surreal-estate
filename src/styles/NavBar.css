.navbar {
  margin: 0;
  padding: 0;
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo {
  width: 200px;
  height: 65px;
}

.navbar-links {
  display: flex;
  text-align: center;
  list-style-type: none;
  font-weight: 600;
  padding: 0px;
  margin: 0px;
}

.navbar-links-item, .sign-out {
  text-decoration: none;
  text-align: center;
  margin: 15px;
  justify-content: center;
  padding: 5px 10px 5px 10px
}

.selected-link {
  background-color: #5a2c8c;
  border-radius: 5px;
  color: white;
  margin: 15px;
  padding: 5px 10px 5px 10px

}

.sign-out {
  border-radius: 5px;
  background-color: #5a2c8c;
  border: #5a2c8c 1px solid;
  font-family: 'Poppins', sans-serif; 
  font-weight: 600;
  color: white;
  padding: 5px 10px 5px 10px
}
