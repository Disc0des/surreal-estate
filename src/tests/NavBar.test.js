import React from "react";
import { render } from "@testing-library/react";
import NavBar from "../components/NavBar";

describe("View NavBar", () => {
  it("renders navbar when page is loaded", () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Navlinks are visible", () => {
    const { getByText } = render(<NavBar />);
    expect(getByText("Add Property")).toHaveClass("navbar-label");
    expect(getByText("View Properties")).toHaveClass("navbar-label");
  });
});
