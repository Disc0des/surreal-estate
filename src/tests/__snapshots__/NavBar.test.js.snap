// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View NavBar renders navbar when page is loaded 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      class="logo"
      src="estate-logo.png"
    />
    <ul
      class="navbar-links"
    >
      <a
        aria-current="page"
        class="navbar-links-item selected-link"
        href="/"
      >
        <li
          class="navbar-label"
        >
          View Properties
        </li>
      </a>
      <a
        class="navbar-links-item"
        href="/add-property"
      >
        <li
          class="navbar-label"
        >
          Add Property
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
